### ISSUE TITLE
Player cannot shoot enemies

### ISSUE THREAD

**Issue by StevenPolley** [gdm85/wolfengo#9]
I've narrowed this down to line 303 in src/level.go.  It appears that this is returning nil when it should not be.  I don't really have the know-how to troubleshoot this further. 

`nearestMonsterIntersect := findNearestVector2f(nearestMonsterIntersect, collisionVector, lineStart)`

 I've verified nearestMonsterIntersect is not nil when it shouldn't be. I've also began chasing down collisionVector and noticed that the java version had a method monster.getSize() whereas the go version instead accesses the set variables.  It seems to work out to be about the same end result, so I don't think this is it.

lineStart is set in player.go on line 171, then the above problematic function in level.go is called a few lines below that.  Is lineStart supposed to be the position of the mouse?  I'm not quite following as I'm quite new to golang and a hobby programmer at best.

**Comment by gdm85** [comment]
@StevenPolley thanks for looking into this! I have actually always been thinking about this issue, and never could find the solution. I had already found out the problem with `nearestMonsterIntersect`, but the bug was hidden in plain sight for me; now that you opened this issue I went to look again and could finally find it, see commit b639abff14abd488f0e61c95b70e317b24e568ff

The bug could only be a Go-specific pitfall, see https://github.com/gdm85/wolfengo/commit/b639abff14abd488f0e61c95b70e317b24e568ff#diff-121a9b057dd1b9e61a8ff4389427d8a1L303

Basically using `:=` instead of `=` was the issue; a linter could probably have spotted the variable shadowing as well.
